controller/src/main/java/io/pravega/controller/store/stream/tables/StreamTruncationRecord.java

```diff
@@ -9,29 +9,37 @@
  */
 package io.pravega.controller.store.stream.tables;
 
+import com.google.common.base.Preconditions;
 import com.google.common.collect.ImmutableMap;
 import com.google.common.collect.ImmutableSet;
-import lombok.AccessLevel;
-import lombok.AllArgsConstructor;
+import io.pravega.common.ObjectBuilder;
+import io.pravega.controller.store.stream.tables.serializers.StreamTruncationRecordSerializer;
+import lombok.Builder;
+import lombok.Data;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
 
-import java.io.Serializable;
-import java.util.Collections;
+import java.io.IOException;
 import java.util.Comparator;
 import java.util.HashSet;
+import java.util.Map;
 import java.util.Set;
 
 /**
  * Data class for storing information about stream's truncation point.
  */
-@AllArgsConstructor(access = AccessLevel.PACKAGE)
-public class StreamTruncationRecord implements Serializable {
+@Data
+@Slf4j
+public class StreamTruncationRecord  {
+    public static final StreamTruncationRecordSerializer SERIALIZER = new StreamTruncationRecordSerializer();
+
     public static final StreamTruncationRecord EMPTY = new StreamTruncationRecord(ImmutableMap.of(),
-            ImmutableMap.of(), ImmutableSet.of(), ImmutableSet.of());
+            ImmutableMap.of(), ImmutableSet.of(), ImmutableSet.of(), false);
 
     /**
      * Stream cut that is applied as part of this truncation.
      */
-    private final ImmutableMap<Integer, Long> streamCut;
+    private final Map<Integer, Long> streamCut;
```

Comment:
Why are you making these maps and sets mutable?

Reply:
reverted